projects/Mallard/src/download-edition/download-and-unzip.ts

```diff
@@ -74,76 +77,81 @@ const runDownload = async (issue: IssueSummary, imageSize: ImageSize) => {
             Feature.DOWNLOAD,
         )
 
-        const issueDataDownload = await downloadNamedIssueArchive({
-            localIssueId: localId,
-            assetPath: assets.data,
-            filename: 'data.zip',
-            withProgress: false,
-        }) // just the issue json
-
-        await issueDataDownload.promise
-
-        await pushTracking('attemptDataDownload', 'completed', Feature.DOWNLOAD)
-
-        await pushTracking(
-            'attemptMediaDownload',
-            JSON.stringify({ localId, assets: assets[imageSize] }),
-            Feature.DOWNLOAD,
-        )
-
-        const imgDL = await downloadNamedIssueArchive({
-            localIssueId: localId,
-            assetPath: assets[imageSize] as string,
-            filename: 'media.zip',
-            withProgress: true,
-        }) // just the images
-
-        await imgDL.promise
-
-        await pushTracking(
-            'attemptMediaDownload',
-            'completed',
-            Feature.DOWNLOAD,
+        await retry(
+            async () => {
+                // We are not asking for progress update during Data bundle download (to avoid false visual completion)
+                // So, instead we are artificially triggering a small progress update to render some visual change
+                updateListeners(localId, {
```

Comment:
Is this required?

Reply:
When an issue download is trigged it first download the `data` zip file that can take from one second to couple of seconds. During this time users don't see any visual progress (circular progress bar). To fill that gap I have added this `0.5%` artificial progress update so that user can see circular progress bar right after they press the button